### PR TITLE
Add RNA-seq tool

### LINKED ISSUE #17: Add RNA-seq QC tool
A la [RNA-seQC](https://www.broadinstitute.org/cancer/cga/rna-seqc).

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #17.

Needs squash.

### CODE REVIEW

File: quinine-cli/src/main/scala/org/bdgenomics/quinine/cli/RNAMetrics.scala [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+class RNAMetricsArgs extends Args4jBase {
+  @Argument(required = true,
+    metaVar = "READS",
+    usage = "The sequenced RNA-seq dataset.",
+    index = 0)
+  var readPath: String = null
+
+  @Argument(required = true,
+    metaVar = "TRANSCRIPTS",
+    usage = "The descriptions of the transcripts.",
```
Comment (tdanford) [inline quinine-cli/src/main/scala/org/bdgenomics/quinine/cli/RNAMetrics.scala]: Should there be a more elaborate or longer description of what this file, or its format, should be?

File: quinine-core/src/main/scala/org/bdgenomics/quinine/metrics/rna/RNASeqAlignmentCounts.scala [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      intronic + that.intronic,
+      rRNA + that.rRNA)
+  }
+
+  /**
+   * @param totalReads The total number of reads in this dataset.
+   * @return Returns the ratio of exon mapped reads to total reads.
+   */
+  def profileEfficiency(totalReads: Long): Double = {
+    exonic.toDouble / totalReads.toDouble
```
Comment (tdanford) [inline quinine-core/src/main/scala/org/bdgenomics/quinine/metrics/rna/RNASeqAlignmentCounts.scala]: Zero check?

Comment (fnothaft) [inline quinine-core/src/main/scala/org/bdgenomics/quinine/metrics/rna/RNASeqAlignmentCounts.scala]: If someone runs this on an input dataset with no reads:

1. They have bigger problems than a div/0.
2. They'll get an empty reduce upstream of this.

File: quinine-cli/src/main/scala/org/bdgenomics/quinine/cli/RNAMetrics.scala (hunk 1 above)
Comment (fnothaft) [inline quinine-cli/src/main/scala/org/bdgenomics/quinine/cli/RNAMetrics.scala]: Will revise.

File: quinine-core/src/main/scala/org/bdgenomics/quinine/metrics/rna/RNASeqPanelStats.scala [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+      }
+    }
+
+    // compute mean coverage at the ends of transcripts
+    val coverageStatsStartEnd = CoverageStats(targetCoverageSites.flatMap(kv => {
+      val (region, (site, coverage)) = kv
+
+      // demorgan's law up in this
+      // we keep the site if it overlaps with at least one of the start/end of a region
+      // or, in other words, if it doesn't not overlap with all regions
```
Comment (tdanford) [inline quinine-core/src/main/scala/org/bdgenomics/quinine/metrics/rna/RNASeqPanelStats.scala]: I can't parse the third line of this comment :-)

Comment (fnothaft) [inline quinine-core/src/main/scala/org/bdgenomics/quinine/metrics/rna/RNASeqPanelStats.scala]: Are you familiar with DeMorgan's law? If so, it explains that line of the comment.

File: quinine-core/src/main/scala/org/bdgenomics/quinine/metrics/rna/RNASeqPanelStats.scala [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+      if (fragment(ReferenceRegion(region)).forall(r => !r.overlaps(site))) {
+        None
+      } else {
+        Some((site, coverage))
```
Comment (tdanford) [inline quinine-core/src/main/scala/org/bdgenomics/quinine/metrics/rna/RNASeqPanelStats.scala]: Basically, you only return `(site, coverage)` pairs that overlap exactly one end of the fragment (or just the region itself, if it's shorter than the minimum length), is that correct?

File: quinine-core/src/main/scala/org/bdgenomics/quinine/metrics/rna/RNASeqAlignmentCounts.scala (hunk 2 above)
Comment (tdanford) [inline quinine-core/src/main/scala/org/bdgenomics/quinine/metrics/rna/RNASeqAlignmentCounts.scala]: ah.

File: quinine-core/src/main/scala/org/bdgenomics/quinine/metrics/rna/RNASeqPanelStats.scala (hunk 3 above)
Comment (tdanford) [inline quinine-core/src/main/scala/org/bdgenomics/quinine/metrics/rna/RNASeqPanelStats.scala]: I am familiar with boolean logic, I'm just suggesting that the double-negative makes it difficult to parse.

File: quinine-core/src/main/scala/org/bdgenomics/quinine/metrics/rna/RNASeqPanelStats.scala (hunk 4 above)
Comment (fnothaft) [inline quinine-core/src/main/scala/org/bdgenomics/quinine/metrics/rna/RNASeqPanelStats.scala]: Any end of the fragment.

**Pull request merged**